server/store/user_store.go

```diff
@@ -157,6 +169,24 @@ func (s *pluginStore) AtomicStoreUserInIndex(user *User) error {
 	})
 }
 
+func (s *pluginStore) AtomicStoreUserInIndex(user *User) error {
+	return s.AtomicModifyUserIndex(func(userIndex UserIndex) (UserIndex, error) {
+		newUser := &UserShort{
+			MattermostUserID: user.MattermostUserID,
+			RemoteID:         user.Remote.ID,
+			Email:            user.Remote.Mail,
+		}
+
+		for i, u := range userIndex {
+			if u.MattermostUserID == user.MattermostUserID && u.RemoteID == user.Remote.ID {
+				result := append(userIndex[:i], newUser)
+				return append(result, userIndex[i+1:]...), nil
+			}
+		}
+		return userIndex, nil
```

Comment:
0/5 should return an error if it cannot find the user?

Reply:
This is a bug. The user should be appended to the end of the list if not found in the loop.